src/System.Management.Automation/engine/PSClassSearcher.cs

```diff
@@ -296,8 +296,7 @@ private PSClassInfo ConvertToClassInfo(PSModuleInfo module, ScriptBlockAst ast,
 
             foreach (var member in statement.Members)
             {
-                PropertyMemberAst propAst = member as PropertyMemberAst;
-                if (propAst != null)
+                if (member is PropertyMemberAst propAst && (propAst.PropertyAttributes & PropertyAttributes.Hidden) != PropertyAttributes.Hidden)
```

Comment:
> (propAst.PropertyAttributes & PropertyAttributes.Hidden) != PropertyAttributes.Hidden

This can be changed to `!propAst.PropertyAttributes.HasFlag(PropertyAttributes.Hidden)`. The perf issue with `HasFlag` is fixed in 2.1